### PR TITLE
MC33816 dram update

### PR DESCRIPTION
**Pull request by NOx-z**
includes read+write into dram, with initial offsets, and an update Vboost function

### CODE REVIEW

**Comment by rusefi** [comment]
Can one of the admins verify this patch? (meaning the author is not on the white list)

**Comment by rusefi** [comment]
Can one of the admins verify this patch? (meaning the author is not on the white list)

**Comment by rusefi** [comment]
Can one of the admins verify this patch? (meaning the author is not on the white list)

**Comment by rusefi** [comment]
Can one of the admins verify this patch? (meaning the author is not on the white list)

File: firmware/hw_layer/mc33816.cpp [hunk 1]
```diff
@@ -117,6 +110,51 @@ static void setup_spi() {
...
+    // read (MSB=1) at data ram x9 (SCV_I_Hold), and 1 word
+    spi_writew((0x8000 | addr << 5) + 1);
+    readValue = recv_16bit_spi();
+
+    spiUnselect(driver);
+    return readValue;
+}
+
+// Update a single word in Data RAM
+void mcUpdateDram(MC33816Mem addr, unsigned short data) {
```
Comment (mck1117) [inline firmware/hw_layer/mc33816.cpp:137]: perhaps `uint16_t` instead of `unsigned short`

File: firmware/hw_layer/mc33816.cpp [hunk 2]
```diff
@@ -117,6 +110,51 @@ static void setup_spi() {
...
+    spi_writew((addr << 5) + 1);
+    spi_writew(data);
+
+    spiUnselect(driver);
+}
+
+void setBoostVoltage(float volts)
+{
+	// Specifically for Discovery's ~4.5v 5v rail
+	unsigned short data = volts * 3.2;
```
Comment (mck1117) [inline firmware/hw_layer/mc33816.cpp:152]: This isn't an artifact, the correct value from the datasheet is 3.2 counts per volt.

There's a 1/32 divider on the input, then the DAC's output is 9.77mV per LSB.  (1 / 32) / 0.00977 = 3.199 counts per volt.

File: firmware/hw_layer/mc33816.cpp [hunk 3]
```diff
@@ -117,6 +110,51 @@ static void setup_spi() {
...
+    spi_writew(0x0004);
+    // write (MSB=0) at data ram x9 (SCV_I_Hold), and 1 word
+    spi_writew((addr << 5) + 1);
+    spi_writew(data);
+
+    spiUnselect(driver);
+}
+
+void setBoostVoltage(float volts)
+{
```
Comment (mck1117) [inline firmware/hw_layer/mc33816.cpp:150]: curly brace format :(

**Pull request merged**